pkg/machine/qemu/machine.go

```diff
@@ -531,23 +532,58 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 		}
 	}
 
-	// If the qemusocketpath exists and the vm is off/down, we should rm
-	// it before the dial as to avoid a segv
-	if err := v.QMPMonitor.Address.Delete(); err != nil {
+	vlanSocket, err := machineSocket(v.Name, "vlan", "")
+	if err != nil {
 		return err
 	}
-
-	qemuSocketConn, err = machine.DialSocketWithBackoffs(maxBackoffs, defaultBackoff, v.QMPMonitor.Address.Path)
+	err = vlanSocket.Delete()
 	if err != nil {
 		return err
 	}
-	defer qemuSocketConn.Close()
+	isFdVlanVM := false
+	for _, c := range v.CmdLine {
+		if c == fdVlanNetdev() {
+			isFdVlanVM = true
+		}
+	}
 
-	fd, err := qemuSocketConn.(*net.UnixConn).File()
+	forwardSock, forwardState, forwarderProcess, err := v.startHostNetworking(vlanSocket)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to start host networking: %q", err)
+	}
+
+	if isFdVlanVM {
```

Comment:
break this out to a new function ...

Reply:
This doesn't play well. We have to `defer` invocations in this block. The best the body would look after the change is something like:
```go
		fd, qemuSocketConn, err = getVlanFd(vlanSocket.GetPath())
		if qemuSocketConn != nil {
			defer qemuSocketConn.Close()
		}
		if err != nil {
			return nil
		}
		defer fd.Close()
```

It is not compact, neither it is a lot more readable than the current thing. Especially processing one of the output parameters in both error and success cases is looking confusing (at least for a non-golang dev). I can code change this if you insist. Will address the else branch as requested.